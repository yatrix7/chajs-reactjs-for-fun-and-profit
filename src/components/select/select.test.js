import React from 'react'
import renderer from 'react-test-renderer'
import Select from './select'

it('renders correctly with weight', () => {
	const obj = { value: 9 }
	const tree = renderer
		.create(
			<Select
				id={'9'}
				name={'keanumovies'}
				label={'Keanu Movies'}
				value={obj.value}
				options={[
					{ id: 1, name: 'Point Break' },
					{ id: 2, name: 'John Wick' },
					{ id: 3, name: 'The Matrix' }
				]}
				onChange={jest.fn()}
				className={'keanu-movies'}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
