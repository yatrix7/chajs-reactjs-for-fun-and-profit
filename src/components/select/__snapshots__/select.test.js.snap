// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with weight 1`] = `
<div
  className="keanu-movies"
>
  <label
    className="label"
    htmlFor="9"
  >
    Keanu Movies
  </label>
  <select
    className="form-control"
    id="9"
    name="keanumovies"
    onChange={[MockFunction]}
    value={9}
  >
    <option
      value={-1}
    >
      -
    </option>
    <option
      value={1}
    >
      Point Break
    </option>
    <option
      value={2}
    >
      John Wick
    </option>
    <option
      value={3}
    >
      The Matrix
    </option>
  </select>
</div>
`;
